samples/trigger-customizations/README.md

```diff
@@ -0,0 +1,265 @@
+- Create a slow sink for each trigger
+- Observe the results through the logs of the sinks
+
+#### Components
+New components introduced in this demo:
+
+- [event-sender](https://github.com/knative-sandbox/eventing-rabbitmq/tree/main/samples/trigger-customizations/event-sender)
+Is used to send `n` messages to a broker.
+
+- [event-display](https://github.com/knative-sandbox/eventing-rabbitmq/tree/main/samples/trigger-customizations/event-display)
+Is used as a "slow" sink. It will wait for `delay` number of seconds for each event it consumes.
+
+#### Steps
+Execute the following steps/commands from the root of this repo.
+
+#### Create a namespace
+Create a new namespace for the demo. This will make cleanup easier.
+
+```sh
+kubectl apply -f samples/trigger-customizations/100-namespace.yaml
+```
+or
+```sh
+kubectl create ns trigger-demo
+```
+
+####Create a RabbitMQ Cluster
+
+```sh
+kubectl apply -f samples/trigger-customizations/200-rabbitmq.yaml
+```
+or
+```
+kubectl apply -f - << EOF
+apiVersion: rabbitmq.com/v1beta1
+kind: RabbitmqCluster
+metadata:
+  name: rabbitmq
+  namespace: trigger-demo
+spec:
+  replicas: 1
+EOF
+```
+
+#### Create a Broker
+```sh
+kubectl apply -f samples/trigger-customizations/300-broker.yaml
+```
+or
+```sh
+kubectl apply -f - << EOF
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: default
+  namespace: trigger-demo
+  annotations:
+    eventing.knative.dev/broker.class: RabbitMQBroker
+spec:
+  config:
+    apiVersion: rabbitmq.com/v1beta1
+    kind: RabbitmqCluster
+    name: rabbitmq
+EOF
+```
+
+#### Create the 2 Triggers
+
+```sh
+kubectl apply -f samples/trigger-customizations/400-trigger.yaml
+```
+or
+```sh
+kubectl apply -f - << EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: fifo-trigger
+  namespace: trigger-demo
+spec:
+  broker: default
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: fifo-event-display
+      namespace: trigger-demo
+---
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: high-throughput-trigger
+  namespace: trigger-demo
+  annotations:
+    # Value must be between 1 and 1000
+    # A value of 1 RabbitMQ Trigger behaves as a FIFO queue
+    # Values above 1 break message ordering guarantees but can be seen as more performance oriented.
+    rabbitmq.eventing.knative.dev/prefetchCount: "10"
+spec:
+  broker: default
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: high-throughput-event-display
+      namespace: trigger-demo
+EOF
+```
+
+#### Create the 2 Sinks
+NOTE: ko is used here to create the custom images. Ensure `KO_DOCKER_REPO` is set to an accessible repository for the images.
+```sh
+ko apply -f samples/trigger-customizations/500-sink.yaml
+```
+or
+```sh
+ko apply -f - << EOF
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
```

Comment:
Recommend setting minScale to 1 - https://knative.dev/docs/serving/autoscaling/scale-bounds/#lower-bound just to prevent the pod from disappearing during the demo